pkg/gardenlet/controller/seed/seed/reconciler.go

```diff
@@ -65,72 +67,192 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
-	updateConditions ...gardencorev1beta1.Condition,
-) error {
-	patch := client.StrategicMergeFrom(seed.DeepCopy())
+func (r *Reconciler) reportProgress(log logr.Logger, seed *gardencorev1beta1.Seed) flow.ProgressReporter {
+	return flow.NewImmediateProgressReporter(func(ctx context.Context, stats *flow.Stats) {
+		patch := client.MergeFrom(seed.DeepCopy())
 
-	seed.Status.Conditions = v1beta1helper.MergeConditions(seed.Status.Conditions, updateConditions...)
-	seed.Status.ObservedGeneration = seed.Generation
+		if seed.Status.LastOperation == nil {
+			seed.Status.LastOperation = &gardencorev1beta1.LastOperation{}
+		}
+		seed.Status.LastOperation.Description = flow.MakeDescription(stats)
+		seed.Status.LastOperation.Progress = stats.ProgressPercent()
+		seed.Status.LastOperation.LastUpdateTime = metav1.NewTime(r.Clock.Now().UTC())
+
+		if err := r.GardenClient.Status().Patch(ctx, seed, patch); err != nil {
+			log.Error(err, "Could not report reconciliation progress")
+		}
+	})
+}
+
+func (r *Reconciler) updateStatusOperationStart(ctx context.Context, seed *gardencorev1beta1.Seed, operationType gardencorev1beta1.LastOperationType) error {
+	var (
+		now         = metav1.NewTime(r.Clock.Now().UTC())
+		description string
+	)
+
+	switch operationType {
+	case gardencorev1beta1.LastOperationTypeReconcile:
+		description = "Reconciliation of Seed cluster initialized."
+	case gardencorev1beta1.LastOperationTypeDelete:
+		description = "Deletion of Seed cluster in progress."
+	}
+
+	seed.Status.LastOperation = &gardencorev1beta1.LastOperation{
+		Type:           operationType,
+		State:          gardencorev1beta1.LastOperationStateProcessing,
+		Progress:       0,
+		Description:    description,
+		LastUpdateTime: now,
+	}
 	seed.Status.Gardener = r.Identity
+	seed.Status.ObservedGeneration = seed.Generation
 	seed.Status.ClientCertificateExpirationTimestamp = r.ClientCertificateExpirationTimestamp
-	seed.Status.KubernetesVersion = &seedVersion
+	seed.Status.KubernetesVersion = pointer.String(r.SeedClientSet.Version())
+
+	// Initialize capacity and allocatable
+	var capacity, allocatable corev1.ResourceList
+	if r.Config.Resources != nil && len(r.Config.Resources.Capacity) > 0 {
+		capacity = make(corev1.ResourceList, len(r.Config.Resources.Capacity))
+		allocatable = make(corev1.ResourceList, len(r.Config.Resources.Capacity))
+
+		for resourceName, quantity := range r.Config.Resources.Capacity {
+			capacity[resourceName] = quantity
+			allocatable[resourceName] = quantity
+
+			if reservedQuantity, ok := r.Config.Resources.Reserved[resourceName]; ok {
+				allocatableQuantity := quantity.DeepCopy()
+				allocatableQuantity.Sub(reservedQuantity)
+				allocatable[resourceName] = allocatableQuantity
+			}
+		}
+	}
 
 	if capacity != nil {
 		seed.Status.Capacity = capacity
 	}
-
 	if allocatable != nil {
 		seed.Status.Allocatable = allocatable
 	}
 
-	return c.Status().Patch(ctx, seed, patch)
+	return r.GardenClient.Status().Update(ctx, seed)
```

Comment:
Why was the call changed to `Update` only here?

Reply:
This is also what we do in the `startOperation` functions in other controllers, I just "replicated" the same behaviour:

https://github.com/gardener/gardener/blob/c62e0bdaf9e112ea3c1bf5d528f196f5fc382f6e/pkg/gardenlet/controller/shoot/shoot/reconciler.go#L483-L608
https://github.com/gardener/gardener/blob/c62e0bdaf9e112ea3c1bf5d528f196f5fc382f6e/pkg/gardenlet/controller/shoot/shoot/reconciler.go#L483-L608